calculator.js

```diff
@@ -86,61 +89,72 @@ function reassignedVariables(operator) {
+      reassignedVariables(calculator.nextOperator);
+      showDisplay(calculator.term, calculator.total);
+    } else {
+      calculator.nextOperator = value;
+      calculator.term += calculator.nextOperator;
+      calculator.total = calculation(calculator.total, calculator.current);
+      reassignedVariables(calculator.nextOperator);
+      showDisplay(calculator.term, calculator.total);
+    }
   }
 };
 /*
  *PRE-CONDITIONS: This function gets information from global variables filled previously.
  *POST-CONDITIONS: It does three actions when it activates by event listener. 
  */
 const doActions = function (event) {
-  const character = event.target.value; 
-  if(character === EQUAL) {
-    calculator.total = calculation(calculator.total, calculator.current);
-    livesResult.innerHTML = calculator.total;
-    display.innerHTML = "0";   
-    calculator.current = 0;
-  }
-  if (character === CLEAR) {
-    calculator.current = "";
-    calculator.operator = "";
-    calculator.nextOperator = "";
-    calculator.term = "";
-    calculator.total = 0;
-    display.innerHTML = 0;
-    livesResult.innerHTML = 0;
-  }
-  if (character === DELETE) {
-    const firstTerm = calculator.term;
-    const number = calculator.current;
-    calculator.term = firstTerm.slice(0,-1);
-    calculator.current = number.slice(0,-1);
-    livesResult.innerHTML = calculator.current;
-
-    if(calculator.current.length === 0) {
+  const character = event.target.value || event.key; 
+  console.log(character);
+  let action = character === CLEAR || character === EQUAL || character === DELETE;
+  if (action) {
+    if(character === EQUAL) {
+      calculator.total = calculation(calculator.total, calculator.current);
+      livesResult.innerHTML = calculator.total;
+      display.innerHTML = "0";   
+      calculator.current = 0;
+    }
+    if (character === CLEAR) {
+      calculator.current = "";
+      calculator.operator = "";
+      calculator.nextOperator = "";
+      calculator.term = "";
+      calculator.total = 0;
+      display.innerHTML = 0;
       livesResult.innerHTML = 0;
     }
+    if (character === DELETE) {
+      const firstTerm = calculator.term;
+      const number = calculator.current;
+      calculator.term = firstTerm.slice(0,-1);
+      calculator.current = number.slice(0,-1);
+      livesResult.innerHTML = calculator.current;
+  
+      if(calculator.current.length === 0) {
+        livesResult.innerHTML = 0;
+      }
+    }
   }
 };
 numbers.forEach(function(input) {
-  input.addEventListener("click", saveNumber)
+  input.addEventListener("click", saveNumber);
+  document.addEventListener("keypress",saveNumber);
 });
 operators.forEach(function(operators) {
   operators.addEventListener("click", operate)
+  document.addEventListener('keypress', operate);
 });
 actions.forEach(function(actions) {
   actions.addEventListener("click", doActions)
-});
\ No newline at end of file
+  document.addEventListener('keypress', doActions);
```

Comment:
This is inefficient, you are attaching more event listeners than needed. Attach only one event listener to the `document` object. Also, use `keydown` instead of `keypress`. Search for the difference.